pkg/server/proto/server.proto

```diff
@@ -485,6 +485,12 @@ service Waypoint {
 
   // List releases for a given application.
   rpc UI_ListReleases(UI.ListReleasesRequest) returns (UI.ListReleasesResponse);
+
+  //List UI events for a given application.
+  rpc UI_ListEvents(UI.ListEventsRequest) returns (UI.ListEventsResponse);
+
+  // List UI event schemas (primarily for ensuring types flow through to swagger, then generated by OpenAPI).
```

Comment:
Could you elaborate a little more on what we expect the frontend (or other api clients) to use this for?

Also likely needs an entry in `gateway.yml`

Reply:
@izaaklauer We don't expect anything to call this, but by defining an RPC that uses `UI.EventBuild / UI.EventDeployment / UI.EventRelease` protos, protoc-gen-swagger picks them up when generating the swagger. The swagger is then used when the frontend by openapi-generator and will generate TypeScript types for `UI.EventBuild / UI.EventDeployment / UI.EventRelease`.

We can then use these on the frontend when unmarshaling the untyped EventBundle.EventData field.

Reply:
Thanks @andrew-hashicorp for elaborating. I updated the description comment line above to read: `  // List possible event data types.` per my discussion with Andrew. LMK if yall feel like that could be better explained.

Reply:
Ah interesting. I suppose it makes sense that the types wouldn't make it all the way through to the final js client if it isn't used by an rpc here.

Can we add that as a comment on this RPC then, and rename it to something that makes it clear potential api callers can ignore this? Something like